### PR TITLE
Feature/local storage

### LINKED ISSUE #15: Add and remove from local storage
## User Story

`As a user, in order to view favorites after I refresh the page or view the rockets page then return to the main page, when I click the favorite icon on an article and then click the view favorites button, I should see the articles I've favorited.`

`As a user, in order to delete a favorited article, when I click the favorite icon a second time, I should see only the articles that I have still favorited.  I should also be able to refresh the page and see the same set of articles in the view favorites view.`

## Acceptance Criteria:

 - [x] `As a developer, I click a favorite icon on an article and the article is saved to my favorites even after I refresh the page or view the rockets page.`
 - [x] `As a developer, I click the favorite icon on an article for a second time and the article is removed from view and when the page is refreshed, only the articles currently saved are displayed in the favorites view.`

### PR DESCRIPTION
**Pull request by madhaus4**
# Description
Implements saving, retrieving and deleting a favorited article from local storage

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?
- [x] Use console.log()
- [x] Dev tools / react dev tools
- [x] View in local host
- [x] Verify acceptance criteria (if applicable)

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

closes #15

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ function App() {
       .then(data => setRockets(data))
       .catch(error => setError(error))
     getData('v5', 'launches')
-      .then(data => setLaunchImages(data[19].links.flickr.original, data[22].links.flickr.original))
+      .then(data => setLaunchImages(data[19].links.flickr.original, data[21].links.flickr.original))
```
Comment (Meggs625) [inline src/components/App/App.js:24]: Too bad this didn't work...

File: src/components/HistoricalContainer/HistoricalContainer.js [hunk 2]
```diff
@@ -49,11 +70,7 @@ function HistoricalContainer({ theHistory, launchImages }) {
     <section className='historical-cards-section'>
       <button 
         onClick={() => toggleDisplay()}
-        >View Favorites
-      </button>
-      <button
```
Comment (Meggs625) [inline src/components/HistoricalContainer/HistoricalContainer.js:54]: Beautiful refactor

File: src/components/HistoricalContainer/HistoricalContainer.js [hunk 3]
```diff
@@ -49,11 +70,7 @@ function HistoricalContainer({ theHistory, launchImages }) {
     <section className='historical-cards-section'>
       <button 
         onClick={() => toggleDisplay()}
-        >View Favorites
-      </button>
-      <button 
-        onClick={() => toggleDisplay()}
-        >View All
+        >{isFavorited ? 'View All' : 'View Favorites'}
```
Comment (saberg1) [inline src/components/HistoricalContainer/HistoricalContainer.js:73]: Nice implementation of a ternary to DRY the code up.

File: src/components/HistoricalContainer/HistoricalContainer.js [hunk 4]
```diff
@@ -20,14 +20,35 @@ function HistoricalContainer({ theHistory, launchImages }) {
...
+  }  
+
+  const retrieveFavoritesFromStorage = () => {
+    const keys = Object.keys(localStorage).map(element => {
+      return JSON.parse(localStorage.getItem(element))
+    })
+    setFavoritesList(keys)
+  }
+
+  const removeFavoriteFromStorage = (ID) => {
```
Comment (Meggs625) [inline src/components/HistoricalContainer/HistoricalContainer.js:44]: Great looking code here - throw back to the Mod 1

**Review by Meggs625 — APPROVED** [review]

**Pull request merged**